src/odemis/driver/test/xt_client_test.py

```diff
@@ -1120,6 +1120,18 @@ def test_beamlet_index_VA(self):
         self.assertEqual(current_value, self.microscope.get_beamlet_index())
         self.assertEqual(current_value, self.scanner.beamletIndex.value)
 
+    def test_immersion_VA(self):
+        current_value = self.scanner.immersion.value
+
+        for new_val in (not current_value, current_value):
```

Comment:
Nice implementation to set back the original value after trying the other than original value.

However, after the presentation of Thera (and via `import this`) in the Zen of python i found rule 2 stating: _Explicit is better than implicit._ I don't know if setting back the original value is done very explicitly here.
On the other had rule 1 states: _Beautiful is better than ugly._ And since beauty is a matter of taste....

Anyhow if @tepals doesn't mind this implementation is ok

Reply:
I can add a comment to explicit state the trick.